packages/react/src/tree/TreeItemToggle.js

```diff
@@ -12,28 +14,33 @@ const TreeItemToggle = forwardRef((inProps, ref) => {
     children,
     disabled,
     onClick: onClickProp,
+    onKeyDown: onKeyDownProp,
     ...rest
   } = useDefaultProps({ props: inProps, name: 'TreeItemToggle' });
   const {
     isExpanded,
     toggleExpansion,
   } = useTreeItem();
-  const styleProps = useTreeItemToggleStyle();
-  const onClick = useCallback((event) => {
+  const preventNodeSelection = useCallback((event) => {
     // Stop event bubbling to prevent the node from being selected
     event.stopPropagation();
+  }, []);
+  const styleProps = useTreeItemToggleStyle();
 
-    toggleExpansion();
-  }, [toggleExpansion]);
+  const { onClick, onKeyDown } = useButtonEventHandlers({
+    disabled,
+    onActivate: () => ensureFunction(toggleExpansion)(),
+  });
 
   const getTreeItemToggleProps = () => ({
     'aria-disabled': ariaAttr(disabled),
     'aria-expanded': ariaAttr(isExpanded),
     disabled,
-    onClick: callEventHandlers(onClickProp, onClick),
+    onClick: callEventHandlers(onClickProp, preventNodeSelection, onClick),
+    onKeyDown: callEventHandlers(onKeyDownProp, preventNodeSelection, onKeyDown),
     ref,
     role: 'button',
-    tabIndex: 0,
+    tabIndex: -1,
```

Comment:
Note: Set `tabIndex` to `-1` because the `TreeItemToggle` should not be focusable with tab navigation